cpp/src/strings/utilities.hpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuda_runtime.h>
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/string_view.cuh>
+
+#include <rmm/thrust_rmm_allocator.h>
+
+namespace cudf
+{
+namespace strings
+{
+namespace detail
+{
+
+/**
+ * @brief Creates a temporary string_view object from a host string.
+ *
+ * @param[in] str Null-terminated, encoded string in CPU memory.
+ * @param[in] stream Stream to execute any device code against.
+ * @return Device object pointer.
+ */
+std::unique_ptr<string_view, std::function<void(string_view*)>>
+    string_from_host( const char* str, cudaStream_t stream=0 );
+
+/**
+ * @brief Creates a strings array from a strings column.
+ * This is useful for doing some intermediate array operations.
+ *
+ * @param strings Strings instance.
+ * @param stream Stream to execute any device code against.
+ * @return Strings array
+ */
+rmm::device_vector<string_view> create_string_array_from_column(
```

Comment:
Nitpick on naming, but `array` means something different from `vector`. Consider replacing references to `array` with `vector`.